src/modules/pm.c

```diff
@@ -59,14 +60,14 @@ static void receive(const msg_t *const msg, UNUSED const void* userdata) {
         case PM_REQ: {
             pm_upd *up = (pm_upd *)MSG_DATA();
             if (VALIDATE_REQ(up)) {
-                on_pm_req(up->new);
+                on_pm_req(up->new, up->old);
```

Comment:
Note: `VALIDATE_REQ` already takes care of rejecting requests if up->new == current state; we can avoid passing the additional param here, reducing the diff.
See: https://github.com/FedeDP/Clight/blob/master/src/pubsub/validations.c#L184